xarray/conventions/coding.py

```diff
@@ -190,18 +192,39 @@ def _infer_time_units_from_diff(unique_timedeltas):
     return 'seconds'
 
 
+def infer_calendar_name(dates):
+    """Given an array of datetimes, infer the CF calendar name"""
+    if np.asarray(dates).dtype == 'datetime64[ns]':
+        return 'proleptic_gregorian'
+    else:
+        try:
+            return np.asarray(dates)[0].calendar
+        except IndexError:
+            return np.asarray(dates).item().calendar
+
+
 def infer_datetime_units(dates):
     """Given an array of datetimes, returns a CF compatible time-unit string of
     the form "{time_unit} since {date[0]}", where `time_unit` is 'days',
     'hours', 'minutes' or 'seconds' (the first one that can evenly divide all
     unique time deltas in `dates`)
     """
-    dates = pd.to_datetime(np.asarray(dates).ravel(), box=False)
-    dates = dates[pd.notnull(dates)]
-    unique_timedeltas = np.unique(np.diff(dates))
+    # There is a test that uses a list of strings as input (not consistent with
+    # the docstring).  Should this be allowed or not?  We could potentially
+    # continue supporting this, but it would require knowledge of the calendar
+    # type to decode the strings into the appropriate datetimes.
+    if np.asarray(dates).dtype == 'datetime64[ns]':
+        dates = pd.to_datetime(np.asarray(dates).ravel(), box=False)
```

Comment:
In the [prior code](https://github.com/pydata/xarray/blob/master/xarray/conventions.py#L199) this line would allow for arrays of strings to be passed to `infer_datetime_units`.  Now that I've added a conditional statement to check if the date can be cast as a Timestamp, this is no longer the case.  Is it important for this function to support string inputs (i.e. should I make those work) or is it OK to remove that support?